src/bots/telegram.ts

```diff
@@ -152,35 +178,13 @@ export default class TelegramBot extends BotClient {
 
     try {
       // Try to get chat
-      const chat = await this.bot.getChat(channel.id).catch((error) => {
-        if (error.code === 'ETELEGRAM') {
-          const response = error.response.body;
-          const errorCode = response.error_code;
-          switch (errorCode) {
-            case 403: // Bot is not a member of the channel chat
-              return undefined;
-            default:
-              throw error;
-          }
-        }
-        this.logger.error(
-          `Failed to get chat to check permissions on channel ${channel.label}:\n${error}`,
-        );
-        throw error;
-      });
+      const chat = await this.bot.telegram.getChat(channel.id);
       // Check for expected chat errors
-      if (!chat) {
-        return undefined;
-      }
+      assertIsDefined(chat, 'Chat not defined in getUserPermissions()');
```

Comment:
What happens to the bot if an error is thrown here?
Does it still run afterwards?

Reply:
Yes, this is still wrapped in a try-catch clause